### PR TITLE
Fix accidental truncate after seek on inline files

### PR DESCRIPTION
**Pull request by geky**
The cause was mistakenly setting file->ctz.size directly instead of
file->pos, which file->ctz.size gets overwritten with later in
lfs_file_flush.

Also added better seek test cases specifically for inline files. This
should also catch most of the inline corner cases related to
lfs_file_size/lfs_file_tell.

Should fix https://github.com/ARMmbed/littlefs/issues/172

### CODE REVIEW

File: lfs.c [hunk 1]
```diff
@@ -2548,7 +2548,7 @@ static int lfs_file_flush(lfs_t *lfs, lfs_file_t *file) {
                 }
             }
         } else {
-            file->ctz.size = lfs_max(file->pos, file->ctz.size);
+            file->pos = lfs_max(file->pos, file->ctz.size);
```
Comment (pabigot) [inline lfs.c]: @geky This line is either incorrect or useless, because `file->pos` is overwritten by `pos` on line 2560.  Should it be assigning to `pos`?

Comment (pabigot) [reply lfs.c]: Probably incorrect, since in what's below `file->pos` is used to replace `file.ctz.size`.

Comment (geky) [reply lfs.c]: Hi @pabigot, thanks for raising this. To double check is this fixed in https://github.com/ARMmbed/littlefs/pull/257?

Comment (pabigot) [reply lfs.c]: I didn't touch this material in #257; that fixes a different bug.  I misunderstood what this patch was doing the first couple time I read it.  As far as I know there's nothing wrong left here after this was merged.

Comment (geky) [reply lfs.c]: Ah ok, good to know
